src/adapters/SwivelAdapter.sol

```diff
@@ -0,0 +1,158 @@
+    address public marketPlace;
+    /// @notice mapping that determines if a principal has been paused by the admin
+    mapping(uint8 => bool) public paused;
+    /// @notice flag that allows admin to stop all lending and minting across the entire protocol
+    bool public halted;
+
+    /// @notice protocol specific addresses that adapters reference when executing lends
+    /// @dev these addresses are references by an implied enum; adapters hardcode the index for their protocol
+    address[] public protocolAddressess;
+
+    /// @notice a mapping that tracks the amount of unswapped premium by market. This underlying is later transferred to the Redeemer during Swivel's redeem call
+    mapping(address => mapping(uint256 => uint256)) public premiums;
+
+    /// @notice this value determines the amount of fees paid on loans
+    uint256 public feenominator;
+    /// @notice represents a point in time where the feenominator may change
+    uint256 public feeChange;
+    /// @notice represents a minimum that the feenominator must exceed
+    uint256 public constant MIN_FEENOMINATOR = 500;
+
+    /// @notice maps underlying tokens to the amount of fees accumulated for that token
+    mapping(address => uint256) public fees;
+    /// @notice maps a token address to a point in time, a hold, after which a withdrawal can be made
+    mapping(address => uint256) public withdrawals;
+
+    // Reantrancy protection
+    uint256 private constant _NOT_ENTERED = 1;
+    uint256 private constant _ENTERED = 2;
+    uint256 private _status = _NOT_ENTERED;
+
+    // Rate limiting protection
+    /// @notice maximum amount of value that can flow through a protocol in a day (in USD)
+    uint256 public maximumValue = 250_000e27;
+    /// @notice maps protocols to how much value, in USD, has flowed through each protocol
+    mapping(uint8 => uint256) public protocolFlow;
+    /// @notice timestamp from which values flowing through protocol has begun
+    mapping(uint8 => uint256) public periodStart;
+    /// @notice estimated price of ether, set by the admin
+    uint256 public etherPrice = 2_500;
+
+    function approve(address[] calldata a) external {}
+
+    function lend(bytes calldata d) external returns (uint256) {
+        // Parse the calldata into the arguments
+        (
+            uint256[] memory amounts,
+            Swivel.Order[] memory orders,
+            Swivel.Components[] memory components,
+            address pool,
+            uint256 slippage,
+            bool swapFlag
+        ) = abi.decode(
+                d,
+                (
+                    uint256[],
+                    Swivel.Order[],
+                    Swivel.Components[],
+                    address,
+                    uint256,
+                    bool
+                )
+            );
+
+        // Cache a couple oft-referenced variables
+        address underlying = orders[0].underlying;
```

Comment:
Does anything as far as gas golfing change in context of a proxy?

----

Are you caching to save on gas calling the object, or for stack depth?

Reply:
Nope, gas usage stays the same - instead of reading from the array (costly) we cache into the local variable as the underlying is references many times in this method.